tests/test_workspace.py

```diff
@@ -887,3 +890,13 @@ def test_workspace_poiless(datadir):
 
     assert model.config.poi_name is None
     assert model.config.poi_index is None
+
+
+def test_wspace_unexpected_keyword_argument(simplemodels_model_data):
+    model, data = simplemodels_model_data
+    workspace = pyhf.Workspace.build(model, data)
+
+    spec = json.loads(json.dumps(workspace))
```

Comment:
This approach was taken from `test_wspace_immutable`, but
```python
spec = dict(workspace)
```
works fine as far as I can tell. Is there a reason to not use it? Otherwise I can refactor this.

Reply:
If 

```python
spec = dict(workspace)
```

works then I'm more than happy to go with that.

Reply:
Switched to the conversion via `dict`. I tried to look for possible caveats, and the only thing I came across was [in the documentation of `json.dumps`](https://docs.python.org/3/library/json.html#json.dumps): 

> Note Keys in key/value pairs of JSON are always of the type `str`. When a dictionary is converted into JSON, all the keys of the dictionary are coerced to strings. As a result of this, if a dictionary is converted into JSON and then back into a dictionary, the dictionary may not equal the original one. That is, `loads(dumps(x)) != x` if x has non-string keys.

I don't think that this is a problem here, since the spec is defined in JSON to begin with, and all keys are strings.